### PR TITLE
CRM-20610 : Replace payment details block with editable payment list on 'Edit Contribution' form

### PR DESCRIPTION
**Pull request by monishdeb**
* [CRM-20610: Replace payment details block with editable payment list on 'Edit Contribution' form](https://issues.civicrm.org/jira/browse/CRM-20610)

### CODE REVIEW

File: templates/CRM/Contribute/Form/Contribution.tpl [hunk 1]
```diff
@@ -266,6 +266,12 @@
               <td class="label">{$form.trxn_id.label}</td>
               <td {$valueStyle}>{$form.trxn_id.html} {help id="id-trans_id"}</td>
             </tr>
+            <tr class="crm-contribution-form-block-trxn_date">
+              <td class="label">{$form.trxn_date.label}</td>
+              <td {$valueStyle}>{include file="CRM/common/jcalendar.tpl" elementName=trxn_date}<br />
```
Comment (eileenmcnaughton) [inline templates/CRM/Contribute/Form/Contribution.tpl]: We should be using datepicker - note that if you define the dateType in the xml you can use addField

    $form->addField('birth_date', array('entity' => 'contact'), FALSE, FALSE);


https://github.com/litespeedmarc/civicrm-core/blob/ed0ca2486df5e2d7c40e3b8ace28b066830e3961/CRM/Contact/Form/Edit/Demographics.php#L48-L48

File: CRM/Contribute/BAO/Contribution.php [hunk 2]
```diff
@@ -4161,6 +4160,27 @@ public static function getPaymentInfo($id, $component, $getTrxnInfo = FALSE, $us
           }
           $paidByLabel .= " ({$creditCardType}{$pantruncation})";
         }
+
+        // show payment edit link only for payments done via backoffice form
+        $paymentEditLink = '';
```
Comment (eileenmcnaughton) [inline CRM/Contribute/BAO/Contribution.php]: DIdn't another PR cover this link?

Comment (monishdeb) [inline CRM/Contribute/BAO/Contribution.php]: Yup it was handled in https://github.com/civicrm/civicrm-core/pull/10386 which is closed and replaced by this PR

**Comment by monishdeb** [comment]
As per new changes, screencast about editing a Credit card payment via new payment edit form :  

payment-edit-using-cc

**Comment by monishdeb** [comment]
@eileenmcnaughton I have updated the PR and didn't removed the WIP tag as I want to show you the changes (before passing it to you for QA). Please check the screencast attached above

**Comment by monishdeb** [comment]
Closing it in favor of https://github.com/civicrm/civicrm-core/pull/10774

**Comment by eileenmcnaughton** [comment]
Hmm - we'd better make sure the eventual PR has the screencast

**Pull request closed without merging**